File: flutter/lib/services/notes_service.dart
import 'package:multi_notes/models/models.dart';

class NotesService {
  List<Note> notes = [];

  addNote(Note note) {
    notes.add(note);
  }

  deleteNote(int index) {
    notes.removeAt(index);
  }
}

final notesService = NotesService();

File: android_compose/app/src/main/java/com/fmaldonado/multinotescompose/NotesService.kt
package com.fmaldonado.multinotescompose

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.fmaldonado.multinotescompose.models.Note

object NotesService {

    private val notes = mutableListOf<Note>()
    private val liveNotes = MutableLiveData<List<Note>>()

    fun getNotes() = liveNotes as LiveData<List<Note>>

    fun addNote(note: Note) {
        notes.add(note)
        liveNotes.value = notes
    }

    fun removeNote(index: Int) {
        notes.removeAt(index)
        liveNotes.value = notes
    }

}